app/code/core/Mage/Core/Model/Resource/File/Storage/File.php

```diff
@@ -193,19 +194,12 @@ public function saveFile($filePath, $content, $overwrite = false)
         $filename = basename($filePath);
         $path = $this->getMediaBaseDirectory() . DS . str_replace('/', DS ,dirname($filePath));
 
-        if (!file_exists($path) || !is_dir($path)) {
+        if (!is_dir($path)) {
             @mkdir($path, 0777, true);
         }
 
-        $ioFile = new Varien_Io_File();
-        $ioFile->cd($path);
-
-        if (!$ioFile->fileExists($filename) || ($overwrite && $ioFile->rm($filename))) {
-            $ioFile->streamOpen($filename);
-            $ioFile->streamLock(true);
-            $result = $ioFile->streamWrite($content);
-            $ioFile->streamUnlock();
-            $ioFile->streamClose();
+        if ( ! file_exists($path . DS . $filename) || $overwrite) {
+            $result = @file_put_contents($path . DS . $filename, $content, LOCK_EX);
```

Comment:
iam not a fan of silencing errors, cant we get at least the error and log it/put it on the exception?